internal/controller/workspace/workspace.go

```diff
@@ -252,8 +255,15 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		return managed.ExternalObservation{}, errors.Wrap(err, errOptions)
 	}
 
+	o = append(o, terraform.WithArgs(cr.Spec.ForProvider.PlanArgs))
 	differs, err := c.tf.Diff(ctx, o...)
 	if err != nil {
+		if meta.WasDeleted(mg) {
+			if derr := c.Delete(ctx, mg); derr != nil {
+				return managed.ExternalObservation{}, errors.Wrap(derr, errDiff)
+			}
+			return managed.ExternalObservation{ResourceExists: false}, nil
```

Comment:
@ytsarev - if we don't call Delete directly here, we need to return nil so the reconciler will call it later.   Otherwise it never gets called because of the Observe failure.  I think it's easier to read if we call it explicitly here, but it would get called later by the reconciler as long as we don't return an error when the resource is deleted.

Reply:
@bobh66 do we still need to inject deletion into Observation after implementation of the custom arguments?

Reply:
@ytsarev Yes, the custom arguments can't keep the Observe from failing because it runs terraform plan, which does a refresh.  And since the Observe fails the Delete() function never gets called, so the custom argument to the Destroy command never gets invoked.   The only way to get to the Delete call when Observe fails is to call it within the Observe function.